tests/tracer/test_ext.py

```diff
@@ -23,5 +26,148 @@ def _updateenv(monkeypatch, env):
+    mock_repository_url_output = b"https://github.com/scope-demo/scopeagent-reference-springboot2.git"
+    with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (mock_repository_url_output, b"")
+        extracted_repository_url = git.extract_repository_url()
+
+    assert extracted_repository_url == expected_repository_url
+
+
+def test_git_extract_repository_url_error():
+    """On error, the repository url tag should not be extracted, and should internally raise and log the error."""
+    with pytest.raises(ValueError):
+        with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+            mock_popen.return_value.returncode = -1
+            mock_popen.return_value.communicate.return_value = (b"", b"")
+            git.extract_repository_url()
+
+
+def test_git_extract_commit_message():
+    """Make sure that the git commit message is extracted properly."""
+    expected_msg = "Update README.md"
+    mock_output = b"Update README.md"
+    with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (mock_output, b"")
+        extracted_msg = git.extract_commit_message()
+
+    assert extracted_msg == expected_msg
+
+
+def test_git_extract_commit_message_error():
+    """On error, the commit message tag should not be extracted, and should internally raise and log the error."""
+    with pytest.raises(ValueError):
+        with mock.patch("ddtrace.ext.ci.git.subprocess.Popen") as mock_popen:
+            mock_popen.return_value.returncode = -1
+            mock_popen.return_value.communicate.return_value = (b"", b"")
+            git.extract_commit_message()
+
+
+def test_extract_git_metadata():
+    """Test that extract_git_metadata() sets all tags correctly."""
+
+    def side_effect(author):
+        if author:
+            return "John Doe", "john@doe.com", "2021-02-19T08:24:53Z"
+        return "Jane Doe", "jane@doe.com", "2021-01-19T09:24:53Z"
+
+    expected_tags = {
+        git.REPOSITORY_URL: "https://github.com/scope-demo/scopeagent-reference-springboot2.git",
+        git.COMMIT_MESSAGE: "Update README.md",
+        git.COMMIT_AUTHOR_NAME: "John Doe",
+        git.COMMIT_AUTHOR_EMAIL: "john@doe.com",
+        git.COMMIT_AUTHOR_DATE: "2021-02-19T08:24:53Z",
+        git.COMMIT_COMMITTER_NAME: "Jane Doe",
+        git.COMMIT_COMMITTER_EMAIL: "jane@doe.com",
+        git.COMMIT_COMMITTER_DATE: "2021-01-19T09:24:53Z",
+    }
+
+    with mock.patch("ddtrace.ext.git.extract_repository_url") as extract_repo_url:
+        with mock.patch("ddtrace.ext.git.extract_commit_message") as extract_commit_message:
+            with mock.patch("ddtrace.ext.git.extract_user_info") as extract_user_info:
```

Comment:
I feel like this is so much mocking, there might not be a point in having this test?

Does anyone else have an opinion?